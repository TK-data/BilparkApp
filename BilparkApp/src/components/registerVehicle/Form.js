import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
import { reduxForm, Field } from 'redux-form';
import { getCar } from '../../actions/registerCar';

const styles = StyleSheet.create({
  button: {
    backgroundColor: 'blue',
    color: 'white',
    height: 30,
    lineHeight: 30,
    marginTop: 10,
    textAlign: 'center',
    width: 250,
  },
  container: {
  },
  input: {
    borderColor: 'black',
    borderWidth: 1,
    height: 37,
    width: 250,
  },
});

const renderInput = ({ input: { onChange, ...restInput } }) => {
  return (
    <TextInput
      style={styles.input}
      onChangeText={onChange}
      {...restInput}
      placeholder="VH00000"
    />
  );
};

class Form extends Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
  }

  submit(values) {
    this.props.getCar(values.regnr);
  }

  render() {
    const { handleSubmit } = this.props;

    let main = (
      <View>
        <Text>Registreringsnummer:</Text>
        <Field name="regnr" component={renderInput} />
        <TouchableOpacity onPress={handleSubmit(this.submit)}>
          <Text style={styles.button}>Finn bil</Text>
        </TouchableOpacity>
      </View>
    );


    if (this.props.isLoading) {
      main = (
        <View style={styles.container}>
          <Text>Laster..</Text>
        </View>
      );
    } else if (this.props.hasErrored) {
      main = (
        <View>
          <Text>Registreringsnummeret finnes ikke! Prøv på nytt:</Text>
          <Field name="regnr" component={renderInput} />
          <TouchableOpacity onPress={handleSubmit(this.submit)}>
            <Text style={styles.button}>Finn bil</Text>
          </TouchableOpacity>
        </View>
      );
    } else if (this.props.car) {
      main = (
        <View>
          <Text>Er dette din bil?</Text>
          <Text>{this.props.car}</Text>
        </View>
      );
    }

    return (
      <View style={styles.container}>
        {main}
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    car: state.car,
    isLoading: state.carFetchLoading,
    hasErrored: state.carFetchFailure,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getCar: regnr => dispatch(getCar(regnr)),
  };
};

const FormClass = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Form);

export default reduxForm({
  form: 'getCar', // a unique name for this form
})(FormClass);

