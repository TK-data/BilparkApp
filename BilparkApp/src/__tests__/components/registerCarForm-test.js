import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Form from '../../components/registerVehicle/Form';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  carObject: {
    hasErrored: false,
    isLoading: false,
    car: '',
  },
};


describe('Testing register car form', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <Form />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });

  const wrapper = shallow(
    <Form />,
    { context: { store: mockStore(initialState) } },
  );

  it('Contains a form', () => {
    const render = wrapper.dive().dive().dive();
    expect(render.find('Form').exists());
  });

  it('Can click button and submit value', () => {
    const render = wrapper.dive().dive().dive().dive();
    const submitButton = render.find('TouchableOpacity');
    // submitButton.simulate('press');
  });

  /* it('State is changed when button is clicked', () => {
    const render = wrapper.dive().dive().dive().dive();
    const submitButton = render.find('TouchableOpacity');

    const saveToPhoneSpy = sinon.spy(NativeModules.carActions, 'getCar');
    submitButton.simulate('press');

    expect(saveToPhoneSpy.calledOnce).toBe(true);
  }); */
});
