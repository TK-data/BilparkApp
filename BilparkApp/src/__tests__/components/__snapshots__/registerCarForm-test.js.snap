// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing register car form Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Form(Connect(Form)))
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="getCar"
    getFormState={[Function]}
    initialValues={undefined}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anyTouched": false,
      "array": Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      },
      "arrayInsert": [Function],
      "arrayMove": [Function],
      "arrayPop": [Function],
      "arrayPush": [Function],
      "arrayRemove": [Function],
      "arrayRemoveAll": [Function],
      "arrayShift": [Function],
      "arraySplice": [Function],
      "arraySwap": [Function],
      "arrayUnshift": [Function],
      "asyncErrors": undefined,
      "asyncValidating": false,
      "autofill": [Function],
      "blur": [Function],
      "change": [Function],
      "clearAsyncError": [Function],
      "clearFields": [Function],
      "clearSubmit": [Function],
      "clearSubmitErrors": [Function],
      "destroy": [Function],
      "destroyOnUnmount": true,
      "dirty": false,
      "dispatch": [Function],
      "enableReinitialize": false,
      "error": undefined,
      "focus": [Function],
      "forceUnregisterOnUnmount": false,
      "form": "getCar",
      "getFormState": [Function],
      "initialValues": undefined,
      "initialize": [Function],
      "initialized": false,
      "invalid": false,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pristine": true,
      "pure": true,
      "registerField": [Function],
      "registeredFields": undefined,
      "reset": [Function],
      "resetSection": [Function],
      "setSubmitFailed": [Function],
      "setSubmitSucceeded": [Function],
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "startAsyncValidation": [Function],
      "startSubmit": [Function],
      "stopAsyncValidation": [Function],
      "stopSubmit": [Function],
      "submit": [Function],
      "submitFailed": false,
      "submitSucceeded": false,
      "submitting": false,
      "syncErrors": Object {},
      "syncWarnings": Object {},
      "touch": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "triggerSubmit": undefined,
      "unregisterField": [Function],
      "untouch": [Function],
      "updateSyncErrors": [Function],
      "updateSyncWarnings": [Function],
      "updateUnregisteredFields": false,
      "valid": true,
      "validExceptSubmit": true,
      "values": Object {},
      "warning": undefined,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anyTouched": false,
        "array": Object {
          "insert": [Function],
          "move": [Function],
          "pop": [Function],
          "push": [Function],
          "remove": [Function],
          "removeAll": [Function],
          "shift": [Function],
          "splice": [Function],
          "swap": [Function],
          "unshift": [Function],
        },
        "arrayInsert": [Function],
        "arrayMove": [Function],
        "arrayPop": [Function],
        "arrayPush": [Function],
        "arrayRemove": [Function],
        "arrayRemoveAll": [Function],
        "arrayShift": [Function],
        "arraySplice": [Function],
        "arraySwap": [Function],
        "arrayUnshift": [Function],
        "asyncErrors": undefined,
        "asyncValidating": false,
        "autofill": [Function],
        "blur": [Function],
        "change": [Function],
        "clearAsyncError": [Function],
        "clearFields": [Function],
        "clearSubmit": [Function],
        "clearSubmitErrors": [Function],
        "destroy": [Function],
        "destroyOnUnmount": true,
        "dirty": false,
        "dispatch": [Function],
        "enableReinitialize": false,
        "error": undefined,
        "focus": [Function],
        "forceUnregisterOnUnmount": false,
        "form": "getCar",
        "getFormState": [Function],
        "initialValues": undefined,
        "initialize": [Function],
        "initialized": false,
        "invalid": false,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pristine": true,
        "pure": true,
        "registerField": [Function],
        "registeredFields": undefined,
        "reset": [Function],
        "resetSection": [Function],
        "setSubmitFailed": [Function],
        "setSubmitSucceeded": [Function],
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "startAsyncValidation": [Function],
        "startSubmit": [Function],
        "stopAsyncValidation": [Function],
        "stopSubmit": [Function],
        "submit": [Function],
        "submitFailed": false,
        "submitSucceeded": false,
        "submitting": false,
        "syncErrors": Object {},
        "syncWarnings": Object {},
        "touch": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "triggerSubmit": undefined,
        "unregisterField": [Function],
        "untouch": [Function],
        "updateSyncErrors": [Function],
        "updateSyncWarnings": [Function],
        "updateUnregisteredFields": false,
        "valid": true,
        "validExceptSubmit": true,
        "values": Object {},
        "warning": undefined,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
